Plots/Meteograms/NCL_meteo_1.py

```diff
@@ -0,0 +1,291 @@
+                       levels=[-20, -10, 0, 10, 20, 30, 40, 50, 60],
+                       linewidths=0.4,
+                       linestyles='solid',
+                       zorder=4)
+
+# Create lists of coordinates where the contour labels are going to go
+cont2Labels = [(1.71, 3.82), (5.49, 3.23), (9.53, 4.34), (9.27, 3.53),
+               (14.08, 4.81), (19.21, 2.24), (17.74, 1.00), (22.23, 3.87),
+               (12.87, 2.54), (10.45, 6.02), (11.51, 4.92)]
+
+cont3Labels = [(7.5, 6.1), (10.0, 4.58), (19.06, 1.91), (8.68, 0.46),
+               (19.52, 4.80), (16.7, 6.07), (8.62, 5.41), (18.53, 5.46)]
+
+# Manually plot contour labels
+cont2labels = ax1.clabel(contour2,
+                         manual=cont2Labels,
+                         fmt='%d',
+                         inline=True,
+                         fontsize=7)
+cont3labels = ax1.clabel(contour3,
+                         manual=cont3Labels,
+                         fmt='%d',
+                         inline=True,
+                         fontsize=7,
+                         colors='k')
+
+# Set contour label backgrounds white
+[txt.set_bbox(dict(facecolor='white',
+                   edgecolor='none',
+                   pad=.5)) for txt in contour2.labelTexts]
+[txt.set_bbox(dict(facecolor='white',
+                   edgecolor='none',
+                   pad=.5)) for txt in contour3.labelTexts]
+
+# Use the geocat.viz function to set the main title of the plot
+gvutil.set_titles_and_labels(ax1,
+                             maintitle='Meteogram for LGSA, 28/12Z',
+                             maintitlefontsize=18,
+                             ylabel='Pressure (mb)',
+                             labelfontsize=12)
+
+# Determine the labels for each tick on the x and y axes
+yticklabels = np.array(levels, dtype=np.int)
+xticklabels = ['12z', '15z', '18z', '21z', 'Apr29',
+               '03z', '06z', '09z', '12z', '15z', '18z',
+               '21z', 'Apr30', '03z', '06z', '09z',
+               '12z', '15z', '18z', '21z', 'May01', '03z',
+               '06z', '09z', '12z']
+
+# Use the geocat.viz function to set axes limits and ticks
+gvutil.set_axes_limits_and_ticks(ax1,
+                                 xlim=[0, 24],
+                                 ylim=[0, 7],
+                                 xticks=np.arange(0, 25),
+                                 yticks=np.arange(0, 8),
+                                 xticklabels=xticklabels,
+                                 yticklabels=yticklabels)
+
+# Make the ticks on each axis point inwards instead of outwards
+ax1.tick_params(axis="x", direction="in")
+ax1.tick_params(axis="y", direction="in", labelsize=9)
+
+# Add a pad between the y axis label and the axis spine
+ax1.yaxis.labelpad = 4
+
+# Rotate the labels on the x axis so they are vertical
+for tick in ax1.get_xticklabels():
+    tick.set_rotation(90)
+
+# Make an axis to overlay on top of the contour plot
+axin = fig.add_subplot(311)
```

Comment:
Are you sure you need to make a subplot over the contour plot? You should be able to simply call `ax1.barbs()` after calling `ax1.contourf` and `ax.contour`. You may need to specify `zorder` to get the layering right, but I believe there is a simpler solution than this.

If you can do this without making another subplot, can you move the creation of the wind barbs by the code for the contour lines? It will make this more readable.

Reply:
No, because the contour plots are on a plate carree projection, but the x and y values for the barbs are not coordinates. The scale is different, which is also why the x and y tick units in the first plot do not make sense with the contour map.

Reply:
Could you use cartopy's transform points?